### PR TITLE
Add support for setting slack application in deploy

### PR DESCRIPTION
**Pull request by RyanMacG**
Allow for users to set `:slack_application` in their deploy files without changing default behaviour if they don't.

### CODE REVIEW

**Comment by parkr** [comment]
Hey @RyanMacG! Thanks for the pull request. :) What's your motivation for this? Thanks!

**Comment by RyanMacG** [comment]
I'm a previous user of capistrano-slack and enjoyed having the functionality there. Also at work we use the application's URL for `:application` which means all of our deploys have odd links in them as a result. 

I get that it might be a niche feature but it's one I'm fond of

**Comment by parkr** [comment]
Ok, no problem.

There's another line that's using `fetch(:application)` – can you update that as well? We should probably use a helper method like `#slack_application`.

**Comment by parkr** [comment]
What do you think about `slack_app_name` instead of `slack_application`?

**Comment by RyanMacG** [comment]
That sounds good to me, I'll make these changes and push them up

File: lib/capistrano/slack_notify.rb [hunk 1]
```diff
@@ -36,6 +36,10 @@ def slack_webhook_url
       fetch(:slack_webhook_url)
     end
 
+    def slack_application
```
Comment (parkr) [inline lib/capistrano/slack_notify.rb:39]: Should this be consistent with the variable name, i.e. `slack_app_name`?

**Comment by parkr** [comment]
Thanks mate!

**Comment by parkr** [comment]
Released in 1.1.0

**Pull request merged**